src/Common/Enums/ROSNodeModel.cs

```diff
@@ -17,5 +17,34 @@ public class ROSNodeModel
 
         public List<ROSServiceModel> Services { get; set; }
 
+        /// <summary>
+        /// Put all node topics into a single list.
+        /// </summary>
+        /// <param name="nodes"></param>
+        /// <returns></returns>
+        public List<RosTopicModel> GetAllNodeTopics(List<ROSNodeModel> nodes)
+        {
+            List<RosTopicModel> topics = new List<RosTopicModel>();
```

Comment:
If you will use `HashSet` you will not have to worry about the items repeating in the collection. The `HashSet` contains only unique elements.

Reply:
Implemented hashset, thanks!!!